libraries/AP_GSOF/AP_GSOF.cpp

```diff
@@ -169,18 +181,254 @@ void AP_GSOF::parse_vel(uint32_t a)
+    ins_full_nav.latitude = be64todouble_ptr(msg.data, a + 8);
+    ins_full_nav.longitude = be64todouble_ptr(msg.data, a + 16);
+    ins_full_nav.altitude = be64todouble_ptr(msg.data, a + 24);
+    ins_full_nav.vel_n = be32tofloat_ptr(msg.data, a + 32);
+    ins_full_nav.vel_e = be32tofloat_ptr(msg.data, a + 36);
+    ins_full_nav.vel_d = be32tofloat_ptr(msg.data, a + 40);
+    ins_full_nav.speed = be32tofloat_ptr(msg.data, a + 44);
+    ins_full_nav.roll_deg = be64todouble_ptr(msg.data, a + 48);
+    ins_full_nav.pitch_deg = be64todouble_ptr(msg.data, a + 56);
+    ins_full_nav.heading_deg = be64todouble_ptr(msg.data, a + 64);
+    ins_full_nav.track_angle_deg = be64todouble_ptr(msg.data, a + 72);
+    ins_full_nav.ang_rate_x = be32tofloat_ptr(msg.data, a + 80);
+    ins_full_nav.ang_rate_y = be32tofloat_ptr(msg.data, a + 84);
+    ins_full_nav.ang_rate_z = be32tofloat_ptr(msg.data, a + 88);
+    ins_full_nav.acc_x = be32tofloat_ptr(msg.data, a + 92);
+    ins_full_nav.acc_y = be32tofloat_ptr(msg.data, a + 96);
+    ins_full_nav.acc_z = be32tofloat_ptr(msg.data, a + 100);
+
+#if HAL_LOGGING_ENABLED
+    log_ins_full_nav();
+#endif // HAL_LOGGING_ENABLED
+}
+
+void AP_GSOF::parse_ins_rms(uint32_t a)
+{
+    // https://receiverhelp.trimble.com/oem-gnss/gsof-messages-ins-rms.html
+    ins_rms.gps_week = be16toh_ptr(msg.data + a);
+    ins_rms.gps_time_ms = be32toh_ptr(msg.data + a + 2);
+    ins_rms.imu_alignment_status = ImuAlignmentStatus(msg.data[a + 6]);
+    ins_rms.gnss_status = GnssStatus(msg.data[a + 7]);
+    ins_rms.pos_rms_n = be32tofloat_ptr(msg.data, a + 8);
+    ins_rms.pos_rms_e = be32tofloat_ptr(msg.data, a + 12);
+    ins_rms.pos_rms_d = be32tofloat_ptr(msg.data, a + 16);
+    ins_rms.vel_rms_n = be32tofloat_ptr(msg.data, a + 20);
+    ins_rms.vel_rms_e = be32tofloat_ptr(msg.data, a + 24);
+    ins_rms.vel_rms_d = be32tofloat_ptr(msg.data, a + 28);
+    ins_rms.roll_rms_deg = be32tofloat_ptr(msg.data, a + 32);
+    ins_rms.pitch_rms_deg = be32tofloat_ptr(msg.data, a + 36);
+    ins_rms.yaw_rms_deg = be32tofloat_ptr(msg.data, a + 40);
+
+#if HAL_LOGGING_ENABLED
+    log_ins_rms();
+#endif // HAL_LOGGING_ENABLED
+}
+
+void AP_GSOF::parse_llh_msl(uint32_t a)
+{
+    // https://receiverhelp.trimble.com/oem-gnss/gsof-messages-llmsl.html
+    llh_msl.latitude = RAD_TO_DEG_DOUBLE * be64todouble_ptr(msg.data, a);
+    llh_msl.longitude = RAD_TO_DEG_DOUBLE * be64todouble_ptr(msg.data, a + 8);
+    llh_msl.altitude_msl = be64todouble_ptr(msg.data, a + 16);
+    // Assume the model is EGM96 in ArduPilot, but log for BIN analysis.
+    memcpy(llh_msl.model, msg.data + a + 24, sizeof(llh_msl.model));
+    llh_msl.model[sizeof(llh_msl.model) - 1] = '\0';  // ensure null-termination
+
+#if HAL_LOGGING_ENABLED
+    log_llh_msl();
+#endif // HAL_LOGGING_ENABLED
+}
+
+#if HAL_LOGGING_ENABLED
+void AP_GSOF::log_pos_time() const
+{
+
+    // @LoggerMessage: GSPT
+    // @Description: GPS Position Time Metadata from GSOF1
```

Comment:
```suggestion
    // @Description: GSOF GPS Position and Time Metadata
```

Put the cool stuff up the front